pandas/tests/reshape/test_reshape.py

```diff
@@ -454,6 +454,14 @@ def test_dataframe_dummies_preserve_categorical_dtype(self, dtype):
 
             tm.assert_frame_equal(result, expected)
 
+    def test_get_dummies_dont_sparsify_all_columns(self):
+        # GH18914
+        df = DataFrame.from_items([('GDP', [1, 2]), ('Nation', ['AB', 'CD'])])
+        df = get_dummies(df, columns=['Nation'], sparse=True)
+        df2 = df.reindex(columns=['GDP'])
+
+        tm.assert_index_equal(df.index, df2.index)
```

Comment:
assert the frame itself

Reply:
use the sparse arg here (so it handles both cases)

Reply:
I added a few parameterized changes. This one as well as the test in test_combine_concat.py which honestly looks like it needs some cleanup someday.